src/sentry/workflow_engine/migration_helpers/rule_action.py

```diff
@@ -0,0 +1,106 @@
+from typing import Any
+
+from sentry.notifications.models.notificationaction import ActionTarget
+from sentry.workflow_engine.models.action import Action
+from sentry.workflow_engine.typings.notification_action import (
+    ACTION_TYPE_2_INTEGRATION_ID_KEY,
+    ACTION_TYPE_2_TARGET_DISPLAY_KEY,
+    ACTION_TYPE_2_TARGET_IDENTIFIER_KEY,
+    ACTION_TYPE_2_TARGET_TYPE_RULE_REGISTRY,
+    EXCLUDED_ACTION_DATA_KEYS,
+    RULE_REGISTRY_ID_2_INTEGRATION_PROVIDER,
+    SlackDataBlob,
+)
+
+
+def build_slack_data_blob(action: dict[str, Any]) -> SlackDataBlob:
+    """
+    Builds a SlackDataBlob from the action data.
+    Only includes the keys that are not None.
+    """
+    return SlackDataBlob(
```

Comment:
narrowing the type that can exist in the data blob, should have something similar for most integrations, but might be a bit more difficult (& have diminishing gains) for sentry apps and such